### DISCUSSION TITLE
course/starbook/register-optimization

### DISCUSSION THREAD

**Discussion by giscus** [gogf/gf-site#119] [General]
# course/starbook/register-optimization

注册流程存在的问题及优化方案，包括接口参数校验、禁止重复用户名、密码加密方法和函数参数优化等内容。利用GoFrame框架进行参数校验，包含用户名密码基本验证，并应用i18n实现多语言支持。通过设置数据库唯一索引，确保用户唯一性，同时采用MD5加密保障密码安全，提升系统安全和易用性。

https://goframe.org/course/starbook/register-optimization

**Comment by walkmiao** [comment]
为什么我把 i18n 下的 cn 拷贝过去，校验还是提示的英文啊

**Reply by oldme-git** [reply]
启用CN了吗。
`g.I18n().SetLanguage("zh-CN")`

**Reply by canyonwan** [reply]
我也遇到了一样的问题,
main.go文件代码如下: 

```
func main() {
	g.I18n().SetLanguage("zh-CN")

	cmd.Main.Run(gctx.GetInitCtx())
}
```

**Comment by Muprprpr** [comment]
这里教学文档中`Register函数优化`部分 internal/logic/users/users_register.go文件修改中，是不是将import ”star/internal/model“ 替换为	"star/internal/model/do"更合适呢？

这个地方毕竟model并非一个包，而且这里修改后其实本质上依旧用的是do包

**Comment by sjg20010414** [comment]
type RegisterInput struct 的定义放入独立的 internal/model/users.go 业务输入输出模型文件为好。文中的做法直接把它塞入业务逻辑文件，那么代码就不能改成使用抽象 service 接口的方式了。这个应用本身未使用service抽象层，所以可以这么干。
